### PR TITLE
feat(const-oid): add the AssociatedOid trait

### PR DESCRIPTION
**Pull request by npmccallum**
Signed-off-by: Nathaniel McCallum <nathaniel@profian.com>

### CODE REVIEW

File: const-oid/src/lib.rs [hunk 1]
```diff
@@ -29,6 +29,12 @@ pub use crate::{
 use crate::encoder::Encoder;
 use core::{fmt, str::FromStr};
 
+/// Identifies a type with a given OID.
+pub trait Identified {
```
Comment (tarcieri) [inline const-oid/src/lib.rs:33]: I can't think of a better name myself, but I think this trait would be clearer if it had `Oid` somewhere in the name.

Comment (npmccallum) [reply const-oid/src/lib.rs:33]: Object Identifier Identified? :)

Comment (tarcieri) [reply const-oid/src/lib.rs:33]: `Identified` is an extremely vague name for a trait

Comment (npmccallum) [reply const-oid/src/lib.rs]: @tarcieri How about the name `AssociatedOid` which I just pushed?

Comment (tarcieri) [reply const-oid/src/lib.rs]: Sounds good, thanks

**Pull request merged**